caterva2/services/sub.py

```diff
@@ -545,6 +594,30 @@ async def fetch_data(
                                        media_type='application/octet-stream')
 
 
+@app.get('/api/chunk/{path:path}')
+async def get_chunk(
+    path: pathlib.PosixPath,
+    nchunk: int,
+    user: db.User = Depends(current_active_user),
+):
+    abspath, _ = abspath_and_dataprep(path, user=user)
+    lock = locks.setdefault(path, asyncio.Lock())
+    async with lock:
+        if user and path.parts[0] == '@scratch':
+            container = open_b2(abspath)
+            schunk = getattr(container, 'schunk', container)
+            if 'LazyArray' in schunk.meta:
+                # TODO: Support this
+                raise NotImplementedError
```

Comment:
This can be done later, in another PR, right?

Reply:
Yes, we should create a function in blosc2 that given a `nchunk` and a container returns the corresponding slice, and then call `eval` with that slice to later get the compressed chunk with data.